utils/near-cache/src/lib.rs

```diff
@@ -10,37 +10,46 @@ pub struct SyncLruCache<K, V> {
 
 impl<K, V> SyncLruCache<K, V>
 where
-    K: Hash + Eq + PartialEq + Clone,
+    K: Hash + Eq,
     V: Clone,
 {
-    /// Creates new cache with `capacity`.
-    pub fn new(capacity: usize) -> Self {
-        Self { inner: Mutex::new(LruCache::<K, V>::new(capacity)) }
+    /// Creats a new `LRU` cache that holds at most `cap` items.
+    pub fn new(cap: usize) -> Self {
+        Self { inner: Mutex::new(LruCache::<K, V>::new(cap)) }
     }
 
-    /// Gets value for `key`
-    pub fn get_or_insert(&self, key: K, f: impl FnOnce(&K) -> V) -> V
-    where
-        V: Clone,
-    {
-        if let Some(result) = self.get(&key) {
-            return result;
-        }
-        let val = f(&key);
-        let val_clone = val.clone();
-        self.inner.lock().unwrap().put(key, val_clone);
-        val
+    /// Return the value of they key in the cache otherwise computes the value and inserts it into
+    /// the cache. If the key is already in the cache, they gets gets moved to the head of
+    /// the LRU list.
+    pub fn get_or_put(&self, key: K, f: impl FnOnce(&K) -> V) -> V {
```

Comment:
I'd keep the `insert` terminology. I agree that "don't change wrapped API" is a good rule of thumb, as generally that API uses specific idioms for a reason. 

In this specific case though, I think the reason for `put` terminology is mostly historical. The API originated from a very old version of Rust, when the terminology wasn't ironed out at all: https://doc.rust-lang.org/0.12.0/std/collections/lru_cache/struct.LruCache.html. Note `0.12.0` in the URL. 

Today, Rust uses insert everywhere, 

* https://doc.rust-lang.org/stable/std/?search=insert
* https://doc.rust-lang.org/stable/std/?search=put

Reply:
Actaully, I take this back. I didn't realise that we are going to use non-sync version as well, as in https://github.com/near/nearcore/pull/5871. If that's the case, than, yeah, it definitelly is better to align names.